example/messaging/index.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:convert';
+import 'dart:html';
+
+import 'package:firebase/firebase.dart' as fb;
+import 'package:firebase/src/assets/assets.dart';
+import 'package:http/browser_client.dart';
+import 'package:service_worker/window.dart' as sw;
+
+main() async {
+  //Use for firebase package development only
+  await config();
+
+  try {
+    fb.initializeApp(
+        apiKey: apiKey,
+        authDomain: authDomain,
+        databaseURL: databaseUrl,
+        storageBucket: storageBucket,
+        messagingSenderId: messagingSenderId);
+
+    new MessagesApp().showMessages();
+  } on fb.FirebaseJsNotLoadedException catch (e) {
+    print(e);
+  }
+}
+
+class MessagesApp {
+  final InputElement tokenInput;
+  final ButtonElement newNotification;
+  final ParagraphElement payloadData;
+  final InputElement permissionInput;
+
+  MessagesApp()
+      : tokenInput = document.querySelector("#token"),
+        newNotification = document.querySelector('#new_notification'),
+        payloadData = document.querySelector("#payload_data"),
+        permissionInput = document.querySelector("#permission");
+
+  showMessages() async {
+    await sw.register('sw.dart.js');
+    final registration = await sw.ready;
+    final messaging = fb.messaging()
+      ..usePublicVapidKey(vapidKey)
```

Comment:
Put in a check that `vapidKey != null` – and warn the user.

I'm hitting this, but it took some digging to figure out I need to set up my `VAPID_KEY` env value

Reply:
I put the check in usePublicVapidKey and if you call other messaging function that need vapidKey i throw or print a warning ?

Reply:
I'm going to create a parallel pull request which changes the call to `_config[...]` to `_getConfigValue(...)` – if the value is null, it'll throw